python-sdk/nuscenes/tests/test_predict_helper.py

```diff
@@ -0,0 +1,503 @@
+# nuScenes dev-kit.
+# Code written by Freddy Boulton, 2020.
+
+import unittest
+from typing import Dict, List, Any
+from nuscenes.predict import PredictHelper, convert_global_coords_to_local
+import numpy as np
+
+class MockNuScenes:
+    """Mocks the NuScenes API needed to test PredictHelper"""
+
+    def __init__(self, sample_annotations: List[Dict[str, Any]],
+                 samples: List[Dict[str, Any]]):
+
+        self._sample_annotation = {r['token']: r for r in sample_annotations}
+        self._sample = {r['token']: r for r in samples}
+
+    @property
+    def sample_annotation(self,) -> List[Dict[str, Any]]:
+        return list(self._sample_annotation.values())
+
+    def get(self, table_name: str, token: str) -> Dict[str, Any]:
+        assert table_name in {'sample_annotation', 'sample'}
+        return getattr(self, "_" + table_name)[token]
+
+class Test_convert_global_coords_to_local(unittest.TestCase):
+
+    def setUp(self):
+        along_pos_x = np.zeros((5, 2))
+        along_pos_y = np.zeros((5, 2))
+        along_neg_x = np.zeros((5, 2))
+        along_neg_y = np.zeros((5, 2))
+
+        along_pos_x[:, 0] = np.arange(1, 6)
+        along_pos_y[:, 1] = np.arange(1, 6)
+        along_neg_x[:, 0] = -np.arange(1, 6)
+        along_neg_y[:, 1] = -np.arange(1, 6)
+        self.along_pos_x, self.along_pos_y = along_pos_x, along_pos_y
+        self.along_neg_x, self.along_neg_y = along_neg_x, along_neg_y
+
+        y_equals_x = np.zeros((5, 2))
+        y_equals_x[:, 0] = np.arange(1, 6)
+        y_equals_x[:, 1] = np.arange(1, 6)
+        self.y_equals_x = y_equals_x
+
+    def test_heading_0(self):
```

Comment:
These 6 test methods each have 19-33 lines of redundant code. I suggest to create a separate method.

Reply:
Done!